commit 41caaa08cde60c4fec4bf2e5f9610e2a1b9e6a23
Author: Vinson Lee 
Date:   Thu Feb 5 14:48:53 2015 -0800

    lua: Define LUA_COMPAT_APIINTCASTS for Lua >= 5.3.0 compatibility.
    
    In Lua 5.3.0, luaL_checkint was deprecated.
    
    This patch fixes this build error with Lua 5.3.0.
    
    lua/demux.c: In function ‘vlclua_demux_peek’:
    lua/demux.c:55:5: error: implicit declaration of function ‘luaL_checkint’ [-Werror=implicit-function-declaration]
         int n = luaL_checkint( L, 1 );
         ^
    
    Signed-off-by: Vinson Lee 
    Signed-off-by: Jean-Baptiste Kempf 

diff --git a/modules/lua/vlc.h b/modules/lua/vlc.h
index efd94f1..85c7fc1 100644
--- a/modules/lua/vlc.h
+++ b/modules/lua/vlc.h
@@ -38,6 +38,7 @@
 #include 
 #include 
 
+#define LUA_COMPAT_APIINTCASTS
 #define LUA_COMPAT_MODULE
 #include         /* Low level lua C API */
 #include     /* Higher level C API */
